### PR TITLE
Add telemetry

### LINKED ISSUE #11: Add Telemetry
Add telemetry, similar to that on Zoom.

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Add telemetry to MS Teams Meetings.

The telemetry tracked is:

- Every time a meeting is created
- Every time a meeting is not created because there is another recent meeting
- Every time a meeting creation is forced (even if there is a recent meeting)
- Every time a user sign in into MS Teams Meetings
- Every time a user signs out of MS Teams Meetings

We use telemetry to help improve the quality of Mattermost and to make design decisions for future releases.

This PR will be reviewed by the product team who will confirm which telemetry will actually be used to make those design decisions.

#### Ticket Link
Fix #11

### CODE REVIEW

File: server/telemetry.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+func (p *Plugin) trackConnect(userID string) {
+	p.tracker.TrackUserEvent("connect", userID, map[string]interface{}{})
+}
+
+func (p *Plugin) trackDisconnect(userID string) {
+	p.tracker.TrackUserEvent("disconnect", userID, map[string]interface{}{})
+}
+
+func (p *Plugin) trackMeetingStart(userID, source string) {
+	p.tracker.TrackUserEvent("start_meeting", userID, map[string]interface{}{
```
Comment (jfrerich) [inline server/telemetry.go:17]: Would `meeting_started` work better to match the other keys?  (ex. meeting_duplicated, meetins_forced)

**Review by jfrerich — APPROVED** [review]
Looks good.  There are a lot of other syncing related files included.

File: server/command.go [hunk 2]
```diff
@@ -113,6 +116,7 @@ func (p *Plugin) handleDisconnect(args []string, extra *model.CommandArgs) (stri
 	if err != nil {
 		return "Failed to disconnect the user, err=" + err.Error(), nil
 	}
+	p.trackDisconnect(extra.UserId)
 	return "User disconnected from MS Teams Meetings.", nil
```
Comment (hanzei) [inline server/command.go:120]: Nit:
```suggestion

	p.trackDisconnect(extra.UserId)
	return "User disconnected from MS Teams Meetings.", nil
```

File: server/telemetry.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package main
+
+const (
+	telemetryStartSourceWebapp  = "webapp"
+	telemetryStartSourceCommand = "command"
+)
```
Comment (hanzei) [inline server/telemetry.go:6]: nit pick: Add a `type` for these sources

File: webapp/junit.xml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuites name="jest tests" tests="1" failures="0" time="1.507">
+  <testsuite name="undefined" errors="0" failures="0" skipped="0" timestamp="2020-08-25T08:40:32" time="1.133" tests="1">
+    <testcase classname=" Plugin id and version are defined" name=" Plugin id and version are defined" time="0.002">
+    </testcase>
+  </testsuite>
+</testsuites>
```
Comment (hanzei) [inline webapp/junit.xml:7]: The starter template does put this file in `webapp/.gitignore`. Should it ready get added to this repo?

**Review by hanzei — CHANGES_REQUESTED** [review]
One question about `webapp/junit.xml`

**Review by hanzei — APPROVED** [review]

**Review by aaronrothschild — APPROVED** [review]
LGTM

**Pull request merged**